### PR TITLE
feat: support user config htmlGenerating

### PR DESCRIPTION
**Pull request by ClarkXia**
- [x] support user config htmlGenerating to control the generation of html
- [x] do not render document when template is set to be `false` in plugin pha

### CODE REVIEW

File: packages/ice/src/commands/build.ts [hunk 1]
```diff
@@ -142,13 +142,17 @@ const build = async (
       distType,
       prependCode,
     },
+    htmlGenerating,
   } = userConfig;
   let renderMode: RenderMode;
   if (ssg) {
     renderMode = 'SSG';
+    if (!htmlGenerating) {
+      logger.warn('ssg depens on htmlGenerating, ssg will not work when htmlGenerating is set to false.');
```
Comment (luhc228) [inline packages/ice/src/commands/build.ts:151]: depens -> depends

File: packages/ice/src/types/userConfig.ts [hunk 2]
```diff
@@ -62,6 +62,11 @@ export interface UserConfig {
   dropLogLevel?: 'trace' | 'debug' | 'log' | 'info' | 'warn' | 'error';
   minify?: boolean | 'swc' | MinifyOptions;
   compileDependencies?: boolean | string[] | RegExp[];
+  /**
+   * HTML will not be generated when build, If it is false.
+   * @see https://v3.ice.work/docs/guide/basic/config#htmlgenerating
```
Comment (luhc228) [inline packages/ice/src/types/userConfig.ts:67]: 这个文档是已经有了么？

File: packages/ice/src/commands/build.ts (hunk 1 above)
Comment (luhc228) [reply packages/ice/src/commands/build.ts:151]: ssg -> SSG

File: packages/ice/src/commands/build.ts [hunk 3]
```diff
@@ -148,7 +148,7 @@ const build = async (
   if (ssg) {
     renderMode = 'SSG';
     if (!htmlGenerating) {
-      logger.warn('ssg depens on htmlGenerating, ssg will not work when htmlGenerating is set to false.');
+      logger.warn('SSG depends on htmlGenerating, ssg will not work when htmlGenerating is set to false.');
```
Comment (luhc228) [inline packages/ice/src/commands/build.ts:151]: ssg will -> SSG will

也一块改下吧

File: packages/ice/src/types/userConfig.ts (hunk 2 above)
Comment (ClarkXia) [reply packages/ice/src/types/userConfig.ts:156]: 已添加

File: packages/ice/src/commands/build.ts (hunk 3 above)
Comment (ClarkXia) [reply packages/ice/src/commands/build.ts]: fixed

File: packages/ice/src/commands/build.ts (hunk 1 above)
Comment (ClarkXia) [reply packages/ice/src/commands/build.ts]: fixed

**Review by answershuto — APPROVED** [review]

**Review by luhc228 — APPROVED** [review]

**Pull request merged**